### PR TITLE
Support older ios 13.3 and 13.4 devices

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description

This PR attempts to fix compatibility issues for ios 13.3 and ios 13.4 devices (this mainly refers to iPhones and iPads). Changes: 
- Bump TS target to es2020 as this supported by ios 13.3 and 13.4
- Force `safari10` terser mangle option while preserving the [default webpack config](https://webpack.js.org/configuration/optimization/#optimizationminimize)
- When using the `media.matches` API, fallback to the deprecated `addListener` if found
- Replace the flex `gap` property with margin-top and use of "[lobotomized owl selector](https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/)" pattern
- Updates the webpack and terser dependencies

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts [hunk 1]
```diff
@@ -13,9 +13,28 @@ export function usePrefersReducedMotion() {
...
+        // @ts-expect-error This is a fix for Safari iOS 13.4
+        // This version only supports the deprecated addListener method
+        // addListener accepts a single argument, which is the handler
+        return mediaQueryList.addListener(handler);
+      }
+    };
+
+    const removeListenerCompat = (event: any, handler: any) => {
+      if ('addEventListener' in mediaQueryList) {
+        return mediaQueryList.addEventListener(event, handler);
```
Comment (dimkl) [inline packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts:29]: ⛔ i think we want to use `removeEventListener` in this and the above line.

File: packages/clerk-js/tsconfig.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "resolveJsonModule": true,
     "sourceMap": false,
     "strict": true,
-    "target": "ES2019",
+    "target": "ES2020",
```
Comment (dimkl) [inline packages/clerk-js/tsconfig.json:19]: ❓ what issue did this change resolve?

**Review by dimkl — APPROVED** [review]

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export const createFlexGapPropertyIosCompat = (val: string) => {
+  // Resolve gap property compatibility issue on iOS 13
+  // This is not needed for css grid, only for flex
+  // TODO: Test whether we need to make the check more generic
+  if (navigator?.userAgent?.match(/(iphone|ipad).+os.+13.+safari/i)) {
+    return {
+      '& > * + *': {
+        marginTop: val,
+      },
```
Comment (anagstef) [inline packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:9]: ❓ We use the `Flex` primitive with `flex-direction: row`, will this `marginTop` work then?

File: packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts (hunk 1 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts]: This is already fixed, please take another look :)

File: packages/clerk-js/src/ui/elements/FormControl.tsx [hunk 4]
```diff
@@ -213,10 +213,7 @@ export const FormFeedback = (props: FormFeedbackProps) => {
     };
   };
 
-  const FormInfoComponent: Record<
-    FeedbackType,
-    typeof FormErrorText | typeof FormInfoText | typeof FormSuccessText | typeof FormWarningText
-  > = {
+  const FormInfoComponent: Record<FeedbackType, typeof FormErrorText> = {
```
[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/elements/FormControl.tsx:216]: One of the types resolves to any so tslint collapses the type automatically. This is not related to this PR

File: packages/clerk-js/tsconfig.json (hunk 2 above)
Comment (nikosdouvlis) [reply packages/clerk-js/tsconfig.json:19]: This aligns clerk-js with the other packages. No fix here, just an overall improvement

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts (hunk 3 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:9]: Very valid consideration, let me check how many uses we have for flex-direction

File: packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts [hunk 5]
```diff
@@ -13,9 +13,28 @@ export function usePrefersReducedMotion() {
...
+      } else {
+        // @ts-expect-error This is a fix for Safari iOS 13.4
+        // This version only supports the deprecated addListener method
+        // addListener accepts a single argument, which is the handler
+        return mediaQueryList.addListener(handler);
+      }
+    };
+
+    const removeListenerCompat = (event: any, handler: any) => {
+      if ('addEventListener' in mediaQueryList) {
```
Comment (dimkl) [inline packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts:28]: ⛔ i think you should also update this one to `removeEventListener`

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts (hunk 3 above)
Comment (anagstef) [reply packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:9]: Actually, the default direction is `row`

File: packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts (hunk 5 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/hooks/usePrefersReducedMotion.ts:28]: No this check is correct, we want to call the corresponding remove listener method based on feature detection

**Pull request merged**